Shared/Samples/Edit geodatabase with transactions/EditGeodatabaseWithTransactionsView.Model.swift

```diff
@@ -0,0 +1,175 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Foundation
+
+/// The view model for the sample.
+@MainActor
+class Model: ObservableObject {
+    /// The current progress of the running job.
+    @Published private(set) var progress: Progress?
+    
+    /// A map with an oceans basemap.
+    let map: Map = {
+        let map = Map(basemapStyle: .arcGISOceans)
+        map.initialViewpoint = Viewpoint(center: .christmasBay.center, scale: 8e4)
+        return map
+    }()
+    
+    /// The local geodatabase to edit.
+    private(set) var geodatabase: Geodatabase!
+    
+    /// The task for generating and synchronizing the geodatabase with the feature service.
+    private let geodatabaseSyncTask = GeodatabaseSyncTask(url: .saveTheBaySync)
+    
+    /// The parameters for synchronizing the geodatabase and the feature service.
+    private var syncGeodatabaseParameters: SyncGeodatabaseParameters?
+    
+    /// A URL to a temporary directory where the geodatabase file is stored.
+    private let temporaryDirectoryURL = FileManager.createTemporaryDirectory()
+    
+    /// A Boolean value indicating whether a transaction is active on the geodatabase.
+    var hasLocalEdits: Bool {
+        geodatabase?.hasLocalEdits ?? false
+    }
+    
+    deinit {
+        // Removes the temporary directory and geodatabase file.
+        try? FileManager.default.removeItem(at: temporaryDirectoryURL)
+    }
+    
+    /// Sets up geodatabase and map.
+    func setUp() async throws {
```

Comment:
```suggestion
    func setup() async throws {
```

Reply:
"set up" is used as a verb here, so it is two words instead of one.